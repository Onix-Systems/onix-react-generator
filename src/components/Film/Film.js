import React from 'react';
import PropTypes from 'prop-types';
import styles from './sass/Film.module.scss';

export default function Film({ film }) {
  return (
    <div>
      <p>
        {film.Title}
      </p>
      <img className={styles.appBanner} src={film.Poster} alt={film.Title} />
    </div>
  );
}

Film.propTypes = {
  film: PropTypes.shape({
    Title: PropTypes.string.isRequired,
    Poster: PropTypes.string.isRequired
  }).isRequired
};
