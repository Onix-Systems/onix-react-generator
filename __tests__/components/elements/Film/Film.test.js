import React from 'react';
import { render } from '../../../settings/test-utils';
import Film from '../../../../src/components/Film/Film';

describe('<Film />', () => {
  let wrapper;
  const mockFilm = {
    Title: 'mock film title',
    Poster: 'mock film poster'
  };

  beforeEach(() => {
    wrapper = render(<Film film={mockFilm} />);
  });
  test('should render Film component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('Film component should have file title', () => {
    const { getByText } = wrapper;
    expect(getByText(mockFilm.Title)).toBeInTheDocument();
  });
});
