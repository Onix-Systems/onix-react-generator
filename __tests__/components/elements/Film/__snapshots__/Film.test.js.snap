// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Film /> should render Film component 1`] = `
<DocumentFragment>
  <div>
    <p>
      mock film title
    </p>
    <img
      alt="mock film title"
      class="appBanner"
      src="mock film poster"
    />
  </div>
</DocumentFragment>
`;
